src/matlab2tikz.m

```diff
@@ -4327,7 +4300,14 @@ function warnFacetedInterp(m2t)
 
         userInfo(m2t, ['Please change the "arrowHeadSize" option', ...
             ' if the size of the arrows is incorrect.']);
-        arrowHeadSize = sprintf(m2t.ff, abs(m2t.args.arrowHeadSize));
+        arrowHeadSize = abs(m2t.args.arrowHeadSize);
+
+        % Rescale the head size if MaxHeadSize is changed
+        [headSize, isDefault] = getAndCheckDefault('quiver', h, 'MaxHeadSize', 0.2);
```

Comment:
Maybe comment where the `0.2` comes from (something like `defaultHeadSize = 0.2;` could be enough).